File: akka-essentials/Chapter03/ProcessOrderScala/src/main/scala/akka/examples/actors/OrderAggregateActor.scala
package akka.examples.actors

import akka.actor.Actor
import akka.examples.OrderHistory

class OrderAggregateActor extends Actor {

  def receive: Receive = {
    case histo: OrderHistory =>
      println(s"OrderAggregate: Received histo $histo");
    case x =>
      println(s"OrderAggregate: x=$x")
  }

}

File: akka-essentials/Chapter03/ProcessOrder/src/main/java/actors/OrderAggregateActor.java
package akka.examples.actors;

import akka.actor.UntypedAbstractActor;
import akka.examples.OrderHistory;

public class OrderAggregateActor extends UntypedAbstractActor {

    @Override
    public void onReceive(Object message) throws Exception {
        if (message instanceof OrderHistory) {
            OrderHistory histo = (OrderHistory) message;
            System.out.println("OrderAggregate: Received histo");
        }
        else {
            System.out.println("OrderAggregate: "+message.getClass().getName());   
        }
    }

}
